### DISCUSSION TITLE
Read output in real time

### DISCUSSION THREAD

**Discussion by Martin-Gonzalez90** [adamralph/simple-exec#479] [Q&A] [answered]
Hi, I tried very quickly the lib and works fine.
I just wanted to know if there is a way to read the output/error while the process is being executed instead of waiting the process to end.

Thanks

**Comment by adamralph** [comment, marked as answer]
@Martin-Gonzalez90 at the moment, no. The (simple) model is explicitly: start process -> wait for exit -> get output.

I'm currently experimenting with a new API which would provide real time access all the way down to the byte level, but I can't promise anything. It may never see the light of day. For the time being, I suggest you look at other packages, such as [CliWrap](https://github.com/Tyrrrz/CliWrap).

**Reply by Martin-Gonzalez90** [reply]
Thanks for the quick reply :)
